packages/core/lib/events/DomainEventEmitter.ts

```diff
@@ -79,52 +98,19 @@ export class DomainEventEmitter<SupportedEvents extends CommonEventDefinition[]>
         precedingMessageMetadata ?? {},
       )
     }
-
-    if (!data.metadata.correlationId) {
-      data.metadata.correlationId = this.metadataFiller.produceId()
-    }
-
-    const eventTypeName = supportedEvent.publisherSchema.shape.type.value
-
-    if (!this.eventRegistry.isSupportedEvent(eventTypeName)) {
-      throw new InternalError({
-        errorCode: 'UNKNOWN_EVENT',
-        message: `Unknown event ${eventTypeName}`,
-      })
-    }
-
-    const eventHandlers = this.eventHandlerMap[eventTypeName]
-
-    // No relevant handlers are registered, we can stop processing
-    if (!eventHandlers && this.anyHandlers.length === 0) {
-      // @ts-ignore
-      return data
-    }
```

Comment:
Removing this early return because we might not have handlers on `eventHandlerMap`, but we can have them on `anyHandlers`

Reply:
we used to check both in this step. it's no longer possible?

Reply:
You are right! we were checking both, I can put the early return back if you think it makes sense, but I am not sure if it is really useful.
From my point of view, no matter if there are no handlers assigned to the event, event should validated (to avoid errors on old events if we attach a new handler) and spy should have it into account that is what we are doing below, what do you think?

Reply:
fair point!